Aagam_Shah/postgreAssignment/src/controller/student.controller.ts

```diff
@@ -0,0 +1,216 @@
+const db = require('../config/database').pool;
+import { Context } from "vm";
+import * as validation from '../helper/validation';
+
+exports.getById = async (ctx: Context) => {
+  var table = ctx.params.table;
+  var id = ctx.params.id;
+  if(validation.isValidId(id)){
+    try{
+      const response = await db.query(
+      `SELECT * FROM school.${table} where ${table}id = $1`, [id]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid id passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.listAllStudents = async (ctx: Context) => {
+  var page = parseInt(ctx.request.query.page);
+  var size = parseInt(ctx.request.query.size);
+  var isValid = validation.page_validation(page,size);
+
+  if(isValid.message === "true"){
+    let limit = size;
+    let offset = page*size;
+    const response = await db.query(
+      'SELECT * FROM school.student LIMIT $1 OFFSET $2', [limit,offset]
+    );
+    ctx.response.status = 200;
+    ctx.body = response.rows;
+    return;
+  }
+  else{
+    ctx.response.status = isValid.status;
+    ctx.body = isValid.message;
+    return;
+  }
+};
+
+exports.studentOfClass = async (ctx: Context) => {
+  var id = ctx.params.id;
+  if(validation.isValidId(id)){
+    try{
+      const response = await db.query(
+        'select * from school.student where classid = $1', [id]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid id passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.studentOfTeacher = async (ctx: Context) => {
+  var id = ctx.params.id;
```

Comment:
should be teacherId, id would be too gereric

Reply:
done